plugin/openassetio_manager_bal/bal.py

```diff
@@ -112,6 +112,19 @@ def exists(entity_info: EntityInfo, library: dict) -> bool:
     return True
 
 
+def default_entity(trait_set: Set[str], access: str, library: dict):
```

Comment:
Is this intentionally missing a return type annotation? (I can see it on functions in the neighbourhood)

Reply:
Good spot. 

For background, I/we got a bit lazy with type annotations because

* Our Doxygen documentation uses an ancient tool called doxypy (actually hosted on Foundry's GitHub) to combine Python docs in with the C++ API docs. But it's so ancient it gets tripped up by type annotations.  So in the main OpenAssetIO repo there are no Python type annotations :cry: 
* Until recently we supported Python 3.7, which is missing some neat annotation features and would keep failing on CI.

At least the second is no longer true, and the first case doesn't apply outside of the main repo, so I'll try to be more strict on the annotations.

Reply:
Added that particular missing type hint in 02c21f24.  

Looking through the other changes, the other opportunity for type hints is the main `BasicAssetLibrary.py` and the `bal_business_logic_suite.py` tests, neither of which currently make use of type annotations, even though they could.

So I went with laziness/consistency and didn't add annotations there. I'll add an issue though,

Reply:
Added #123 and put "good first issue" / "help wanted" labels on it :wink: 

The "help wanted" label should mean it'll appear on https://clotributor.dev/ - though perhaps that's for the main repo only :thinking: